extensions/lifecycle/conductor/LifecycleConductor.js

```diff
@@ -542,6 +619,12 @@ class LifecycleConductor {
         });
     }
 
+    needsZookeeper() {
+        return this._bucketSource === 'zookeeper' || // bucket list stored in zk
+            this._bucketSource === 'mongodb' || // bucket stream checkpoints in zk
+            this._bucketSource === 'bucketd'; // bucket stream checkpoints in zk
```

Comment:
Little more compact (feel free to ignore):

```suggestion
        return ['zookeeper', 'mongodb', 'bucketd'].includes(this._bucketSource);
```